crates/router/src/core/webhooks/webhook_events.rs

```diff
@@ -166,10 +266,17 @@ async fn determine_identifier_and_get_key_store(
         .await
     {
         // Valid merchant ID
-        Ok(key_store) => Ok((
-            MerchantIdOrProfileId::MerchantId(merchant_id_or_profile_id),
-            key_store,
-        )),
+        Ok(key_store) => {
+            let merchant_account = store
+                .find_merchant_account_by_merchant_id(&merchant_id_or_profile_id, &key_store)
```

Comment:
passing `merchant_id_or_profile_id` as an argument to this function is a bit confusing. You can get merchant_id from key_store.

Reply:
Correct. At this point, `merchant_id_or_profile_id` and `key_store.merchant_id` would have the same value.
I've added a comment to clarify the usage of `merchant_id_or_profile_id`